### PR TITLE
[test] add unit test for PR #17634

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?

Unit test for https://github.com/ray-project/ray/pull/17634
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by iycheng** [comment]
@rkooo567 could you give a look at this one?

File: release/nightly_tests/many_nodes_tests/app_config.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 base_image: "anyscale/ray-ml:pinned-nightly-py37"
-env_vars: {}
+env_vars: {"RAY_gcs_server_rpc_server_thread_num": "8" }
```
Comment (ericl) [inline release/nightly_tests/many_nodes_tests/app_config.yaml:2]: Remove unrelated changes to nightly test?

**Comment by iycheng** [comment]
@rkooo567 @stephanie-wang  is this good for a merge?

File: src/ray/gcs/gcs_server/test/gcs_actor_scheduler_mock_test.cc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  std::shared_ptr<rpc::MockCoreWorkerClientInterface> core_worker_client;
+  std::shared_ptr<rpc::NodeManagerClientPool> client_pool;
+  MockCallback schedule_failure_handler;
+  MockCallback schedule_success_handler;
+  NodeID node_id;
+  WorkerID worker_id;
+};
+
+TEST_F(GcsActorSchedulerTest, KillWorkerLeak1) {
+  auto actor_id = ActorID::FromHex("f4ce02420592ca68c1738a0d01000000");
```
Comment (stephanie-wang) [inline src/ray/gcs/gcs_server/test/gcs_actor_scheduler_mock_test.cc:75]: Could you add some comments to explain the unit tests?

**Review by stephanie-wang — APPROVED** [review]

**Comment by iycheng** [comment]
@stephanie-wang  can we merge this?
